[INFO -- 2018-12-09 07:39:53]
Starting flip_value script
version: 1.1
[INFO -- 2018-12-09 07:39:53]
Initialization strings:
[INFO -- 2018-12-09 07:39:53]
(1) file /tmp/knapsack/ga
[INFO -- 2018-12-09 07:39:53]
initSignal:1
[INFO -- 2018-12-09 07:39:53]
endSignal:8
[INFO -- 2018-12-09 07:39:53]
faultModel:3
[EXCEPTION -- 2018-12-09 07:39:55]
pythonException: Unable to exit loop in struct fields; Exiting wihtout making a bit flip
[EXCEPTION -- 2018-12-09 07:39:55]
pythonException: empty range for randrange() (0,0, 0)
[INFO -- 2018-12-09 07:39:55]
Backtrace BEGIN:
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
#1  0x00007ffff7490f5d in __GI_abort () at abort.c:90
#2  0x00007ffff74d928d in __libc_message (action=action@entry=do_abort, fmt=fmt@entry=0x7ffff7600528 "*** Error in `%s': %s: 0x%s ***\n") at ../sysdeps/posix/libc_fatal.c:181
#3  0x00007ffff74e064a in malloc_printerr (action=<optimized out>, str=0x7ffff76008b0 "double free or corruption (out)", ptr=<optimized out>, ar_ptr=<optimized out>) at malloc.c:5426
#4  0x00007ffff74e273e in _int_free (av=0x7ffff7832c20 <main_arena>, p=<optimized out>, have_lock=0) at malloc.c:4175
#5  0x00007ffff74e744e in __GI___libc_free (mem=<optimized out>) at malloc.c:3145
#6  0x0000555555559846 in __gnu_cxx::new_allocator<int>::deallocate (this=0x7fffffffd820, __p=0x555555786500) at /usr/include/c++/7/ext/new_allocator.h:125
#7  0x0000555555558802 in std::allocator_traits<std::allocator<int> >::deallocate (__a=..., __p=0x555555786500, __n=220) at /usr/include/c++/7/bits/alloc_traits.h:462
#8  0x0000555555557370 in std::_Vector_base<int, std::allocator<int> >::_M_deallocate (this=0x7fffffffd820, __p=0x555555786500, __n=220) at /usr/include/c++/7/bits/stl_vector.h:180
#9  0x0000555555556b87 in std::_Vector_base<int, std::allocator<int> >::~_Vector_base (this=0x7fffffffd820, __in_chrg=<optimized out>) at /usr/include/c++/7/bits/stl_vector.h:162
#10 0x0000555555555ed7 in std::vector<int, std::allocator<int> >::~vector (this=0x7fffffffd820, __in_chrg=<optimized out>) at /usr/include/c++/7/bits/stl_vector.h:435
#11 0x00005555555553fd in crossover (pai=std::vector of length 200, capacity 200 = {...}, mae=std::vector of length 200, capacity 200 = {...}, filho1=std::vector of length 200, capacity 200 = {...}, filho2=std::vector of length 200, capacity 200 = {...}) at knapsackga.cpp:93
#12 0x00005555555556d7 in genetic_algorithm () at knapsackga.cpp:130
#13 0x0000555555555b6c in main () at knapsackga.cpp:165

46	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.

Backtrace END
Fliping value of field: __val
Fliping array at pos: 5
Memory content before bitflip:0xb00xd10xff0xff0xff0x7f0x000x00
Fault Model: Zero bit-flip
base address to flip value: 0x7fffffffd158
address max offset: 8
Memory content after  bitflip:0x000x000x000x000x000x000x000x00
frame name: __GI_raise
symbol name: set
symbol filename: ../sysdeps/unix/sysv/linux/raise.c
symbol line: 39
value: {__val = {0, 0, 0, 140737353902576, 140737488344240, 0, 140737488344472, 2049, 2, 140737351894629, 0, 1925120, 1924580, 1924580, 0, 5}}
value address: 0x7fffffffd130
Type: The type is a structure.
Type sizeof: 128
Field name: __val
Field Type: The type is an array.
Field Type sizeof: 128

[INFO -- 2018-12-09 07:39:55]
Fault Injection Successful after 2.7056100368499756s
[INFO -- 2018-12-09 07:39:55]
event type: exit
[EXCEPTION -- 2018-12-09 07:39:55]
exit code: no exit code available
